modules/rostests/apitests/ntdll/NtQueryObject.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Test for NtQueryObject
+ * COPYRIGHT:   Copyright 2024 Hermès Bélusca-Maïto <hermes.belusca-maito@reactos.org>
+ */
+
+#include "precomp.h"
+
+/* Flags combination allowing all the read, write and delete share modes.
+ * Currently similar to FILE_SHARE_VALID_FLAGS. */
+#define FILE_SHARE_ALL \
+    (FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE)
+
+/* Adapted from kmtests/ntos_ob/ObQuery.c!ObjectNameInformationTests().
+ * Please sync both tests in case you add or remove new features. */
+START_TEST(NtQueryObject)
+{
+    ULONG g_OsVersion =
+        SharedUserData->NtMajorVersion << 8 | SharedUserData->NtMinorVersion;
+
+    NTSTATUS Status;
+    HANDLE DeviceHandle;
+    UNICODE_STRING DeviceName;
+    OBJECT_ATTRIBUTES ObjectAttributes;
+    IO_STATUS_BLOCK IoStatusBlock;
+
+    ULONG BufferSize1, BufferSize2, BufferSize3;
+    struct { OBJECT_NAME_INFORMATION; WCHAR Buffer[MAX_PATH]; } ObjectNameBuffer;
+    PUNICODE_STRING ObjectName = &ObjectNameBuffer.Name;
+
+    /* Test the drive containing SystemRoot */
+#if 0
+    WCHAR NtDeviceName[sizeof(L"\\DosDevices\\?:")];
+    RtlStringCchPrintfW(NtDeviceName, _countof(NtDeviceName),
+                        L"\\DosDevices\\%c:", SharedUserData->NtSystemRoot[0]);
+#else
```

Comment:
Why #if 0'd?

Reply:
It was to indicate the slightly less hackish way to make the string (compare that with what I do in the else case, where I hardcode the index where I insert the drive letter).

Reply:
And the less hackish way is not working, or not preferred for some reason?

Reply:
Both ways work, it's just that the currently-enabled one I call "more hackish" (because of the hardcoded letter offset) takes less instructions to run than the "less hackish" one.